community/modules/compute/htcondor-execute-point/variables.tf

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  description = "Project in which the HTCondor execute points will be created"
+  type        = string
+}
+
+variable "region" {
+  description = "The region in which HTCondor execute points will be created"
+  type        = string
+}
+
+variable "zone" {
+  description = "The default zone in which resources will be created"
+  type        = string
+}
+
+variable "deployment_name" {
+  description = "HPC Toolkit deployment name. HTCondor cloud resource names will include this value."
+  type        = string
+}
+
+variable "labels" {
+  description = "Labels to add to HTConodr execute points"
+  type        = map(string)
+}
+
+variable "machine_type" {
+  description = "Machine type to use for HTCondor execute points"
+  type        = string
+  default     = "n2-standard-4"
+}
+
+variable "startup_script" {
+  description = "Startup script to run at boot-time for HTCondor execute points"
+  type        = string
+  default     = null
+}
+
+variable "network_storage" {
+  description = "An array of network attached storage mounts to be configured"
+  type = list(object({
+    server_ip     = string,
+    remote_mount  = string,
+    local_mount   = string,
+    fs_type       = string,
+    mount_options = string
+  }))
+  default = []
+}
+
+variable "image" {
+  description = "HTCondor execute point VM image"
+  type = object({
+    family  = string,
+    project = string
+  })
+  default = {
+    family  = "hpc-centos-7"
+    project = "cloud-hpc-image-public"
+  }
+}
+
+variable "service_account" {
+  description = "Service account to attach to HTCondor execute points"
+  type = object({
+    email  = string,
+    scopes = set(string)
+  })
+  default = {
+    email = null
+    scopes = [
+      "https://www.googleapis.com/auth/cloud-platform",
```

Comment:
This could probably be reduced in scope. I had to expand scope on the batch login node to communicate with batch api but barring additional requirements it could probably be something like [this](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/ae70fefab93e9d2d47161bb30fe405ebebd15b3e/community/modules/scheduler/cloud-batch-job/variables.tf#L109-L114)

Reply:
Secret Manager has no corresponding OAuth Scopes. The ability to read project secrets is, also, somewhat surprisingly not part of the Editor role. The HTCondor secret is managed with secret manager so every node needs it.

https://cloud.google.com/secret-manager/docs/accessing-the-api#oauth-scopes